app/src/main/java/org/schabi/newpipe/settings/VideoAudioSettingsFragment.java

```diff
@@ -102,6 +113,49 @@ private void updateSeekOptions() {
             toast.show();
         }
     }
+    // Update resolution options
+    // Defaults to 1080p and removes higher options when "Show higher resolutions" switch is false
+    private void updateResolutions() {
+        final String str1080p60 = "1080p60";
+        final String str1440p = "1440p";
+        final String str2160p = "2160p";
+
+        final ArrayList<String> resolutions = new ArrayList<>(Arrays.asList(getResources()
+                .getStringArray(R.array.resolution_list_description)));
+
+        if (!getPreferenceManager()
+                .getSharedPreferences().getBoolean(getString(R.string.show_higher_resolutions_key),
+                        false)) {
+            if ((defaultResolution.getValue().contains(str1440p)) || (defaultResolution.getValue()
+                    .contains(str2160p))) {
+                defaultResolution.setValue(str1080p60);
+            }
+
+            if ((defaultPopupResolution.getValue().contains(str1440p))
+                    || (defaultPopupResolution.getValue().contains(str2160p))) {
+                defaultPopupResolution.setValue(str1080p60);
+            }
+
+            if ((limitMobileDataUsage.getValue().contains(str1440p))
+                    || (limitMobileDataUsage.getValue().contains(str2160p))) {
+                limitMobileDataUsage.setValue(str1080p60);
+            }
```

Comment:
Then you can do this, also using streams to reduce duplication
```suggestion
            Stream.of(defaultResolution, defaultPopupResolution, limitMobileDataUsage)
                .filter(p -> HIGHER_RESOLUTIONS.contains(p.getValue()))
                .forEach(p -> p.setValue(MAX_NON_HIGHER_RESOLUTION));
```